### PR TITLE
Make chart compatible to Fybrik 0.5.0

### LINKED ISSUE #25: Fix module to be compatible with Fybrik 0.5.0

### PR DESCRIPTION
**Pull request by revit13**
This PR makes the required changes for the chart to be compatible with Fybrik 0.5.0 release.

It contains:

- Updates to the tags used for publishing the chart and docker images to be compatible to Fybrik 0.5.0
- Fixes to README.md
- Fixes to the module charts to work with Fybrik 0.5.0

Closes #25

### CODE REVIEW

File: hack/tag_value.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+# Copyright 2020 IBM Corp.
+# SPDX-License-Identifier: Apache-2.0
+
+: ${RELEASE:=main}
```
Comment (roytman) [inline hack/tag_value.sh]: where do we use RELEASE var ?

File: fybrikapplication.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: app.fybrik.io/v1alpha1
 kind: FybrikApplication
 metadata:
-  name: hello-world-read-module-test
+  name: my-notebook
```
Comment (roytman) [inline fybrikapplication.yaml:4]: why did you rename hello-world-read-module-test to my-notebook

File: fybrikapplication.yaml [hunk 3]
```diff
@@ -12,13 +12,13 @@ spec:
     purpose: Fraud Detection
     role: Data Owner
   data:
-    - dataSetID: "default/medals-winners"
+    - dataSetID: "fybrik-notebook-sample/medals-winners"
```
Comment (roytman) [inline fybrikapplication.yaml:15]: fybrik-notebook-sample ?

File: .github/workflows/push.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        echo "DOCKER_TAGNAME=$version" >> $GITHUB_ENV
+    - name: Helm tool installer
+      uses: Azure/setup-helm@v1
+      with:
+        version: v3.6.3
+    - name: Build docker image 
+      run: make docker-build
+    - name: Docker push
+      run: |
+        echo "${{ secrets.GITHUB_TOKEN }}" | docker login -u "${{ env.DOCKER_USERNAME }}" --password-stdin "${{ env.DOCKER_HOSTNAME }}"
```
Comment (roytman) [inline .github/workflows/push.sh:52]: did we agree to use GITHUB_TOKEN instead of DOCKER_SECRET

File: README.md [hunk 5]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
 
 
-## Deploy Fybrik module
-1. In your module yaml spec (`hello-world-read-module.yaml`):
-    - Change `spec.chart.name` to your chart registry.
-    - Define `flows` and `capabilities` for your module, an example can be found in `hello-world-read-module.yaml`. 
+## Register as a Fybrik module
+
+To register HWRM (Hello Read World Module) as a Fybrik module apply `hello-world-read-module.yaml` to the fybrik-system namespace of your cluster.
+
```
Comment (roytman) [inline README.md:85]: maybe add the kubectl command 
```bash
kubectl apply -f hello-world-read-module.yaml
```
should user explicitly specify namespace for "apply to the fybrik-system namespace" ?
`-n fybrik-system` ?

File: README.md [hunk 6]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
+To install the latest release run:
```
Comment (roytman) [inline README.md]: In order to install the latest release, run:

File: README.md [hunk 7]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
 
-2. Deploy `FybrikModule` in `fybrik-system` namespace:
 ```bash
-kubectl create -f hello-world-read-module.yaml -n fybrik-system
+kubectl apply -f https://github.com/fybrik/hello-world-read-module/releases/latest/download/hello-world-read-module.yaml -n fybrik-system
 ```
-3. Check if `FybrikApplication` successfully deployed:
+
```
Comment (roytman) [inline README.md:90]: command to check the " `FybrikApplication` successfully deployed"

File: README.md [hunk 8]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
+| Fybrik           | HWRM     | Command
+| ---              | ---     | ---
+| 0.5.x            | 0.5.x   | `https://github.com/fybrik/hello-world-read-module/releases/download/v0.5.0/hello-world-read-module.yaml`
+| master           | main    | `https://raw.githubusercontent.com/fybrik/hello-world-read-module/main/hello-world-read-module.yaml`
+
+
+## Deploy and test Fybrik module
+
+Follow this section to deploy and test the module on a single cluster.
```
Comment (roytman) [inline README.md]: maybe "Here is an example how to deploy and test the module on a single cluster.

File: README.md [hunk 9]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
+
+### Before you begin
+
+Install Fybrik using the [Quick Start](https://fybrik.io/v0.5/get-started/quickstart/) guide. This sample assumes the use of the built-in catalog, Open Policy Agent (OPA) and flight module.
+
+> ***Notice: Please follow `version compatbility matrix` section above for deploying the correct version of Fybrik and this module.***
+
+### Deploy Fybrik module
```
Comment (roytman) [inline README.md]: Deploy **this**  module or Deploy the `hello-world-read-module` module

File: README.md [hunk 10]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
+
+Deploy `FybrikModule` in `fybrik-system` namespace:
```
Comment (roytman) [inline README.md]: again, deploy this module in the ...

File: README.md [hunk 11]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
 ```bash
-kubectl get fybrikmodule hello-world-read-module -n fybrik-system
-kubectl describe fybrikmodule hello-world-read-module -n fybrik-system
+kubectl create -f hello-world-read-module.yaml -n fybrik-system
```
Comment (roytman) [inline README.md]: P.S. `kubectl create` will fail if the module exists

File: README.md [hunk 12]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
+  description := "Redact columns tagged as sensitive in datasets tagged with finance = true"
+  input.action.actionType == "read"
+  input.resource.tags.finance
+  column_names := [input.resource.columns[i].name | input.resource.columns[i].tags.sensitive]
+  count(column_names) > 0
+}
+```
 
-You can define OpenPolicyAgent policy to apply them to datasets. You can follow the `Define data access policies` section in [this example](https://fybrik.io/dev/samples/notebook/).
+  In this sample only the policy above is applied. Copy the policy to a file named sample-policy.rego and then run:
```
Comment (roytman) [inline README.md]: why do you need the sentence "In this sample only the policy above is applied." ?

File: README.md [hunk 13]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
+kubectl get job -n fybrik-blueprints
 kubectl get pods -n fybrik-blueprints
 ```
-If you are using the existing `hello-world-read-module.py`, you should see this in the `kubectl logs` of the `fybrik-blueprints` Pod:
+
+If you are using the `hello-world-read-module` image, you should see this in the `kubectl logs` of your completed Pod:
 ```
-$ kubectl logs <fybrik-blueprints pod> -n fybrik-blueprints
+$ kubectl logs my-notebook-default-hello-world-read-module-5f59d75c8f-v7z7p -n fybrik-blueprints
+
```
Comment (roytman) [inline README.md:186]: kubectl logs <fybrik-blueprints pod> -n fybrik-blueprints
e.g.
 kubectl logs my-notebook-default-hello-world-read-module-5f59d75c8f-v7z7p -n fybrik-blueprints

File: README.md [hunk 14]
```diff
@@ -79,87 +79,146 @@ make helm-chart-push
...
-curl -X GET localhost:8000/medals-winners
 ```
-you get the first 10 rows of the medals-winners dataset.
 
 ## Clean
 
 Run the following command to delete the fybrik application:
 ```bash
-kubectl delete FybrikApplication hello-world-read-module-test -n default
+kubectl delete FybrikApplication my-notebook -n default
```
Comment (roytman) [inline README.md:219]: please add in the end reference to the cleanup section from the quick start (it removes the cluster too)

File: fybrikapplication.yaml (hunk 2 above)
Comment (revit13) [reply fybrikapplication.yaml:4]: it results in a shorter pods name...the blueprint controller constructs the name Release name from the app name:  https://github.com/fybrik/fybrik/blob/58d89a7aa9594796b892295a2bf45b68db35d908/manager/controllers/utils/utils.go#L61

File: fybrikapplication.yaml (hunk 3 above)
Comment (revit13) [reply fybrikapplication.yaml:15]: The assets were created in `fybrik-notebook-sample` namespace: (please see the following lines in the README)

`kubectl apply -f sample_assets/assetMedals.yaml -n fybrik-notebook-sample...
`

File: .github/workflows/push.sh (hunk 4 above)
Comment (revit13) [reply .github/workflows/push.sh:52]: This repo does not have a secret named `DOCKER_SECRET` so I used the default which should work.

File: hack/tag_value.sh (hunk 1 above)
Comment (revit13) [reply hack/tag_value.sh]: Removed it. Thanks

File: README.md (hunk 6 above)
Comment (revit13) [reply README.md]: Changed the wording. Thanks

File: README.md (hunk 8 above)
Comment (revit13) [reply README.md]: Changed it. Thanks

File: README.md (hunk 9 above)
Comment (revit13) [reply README.md]: Fixed. Thanks

File: README.md (hunk 10 above)
Comment (revit13) [reply README.md]: Fixed. Thanks

File: README.md (hunk 11 above)
Comment (revit13) [reply README.md]: Changed to `apply`

File: README.md (hunk 12 above)
Comment (revit13) [reply README.md]: I removed it... thanks (its taken from Fybrik site btw)

File: README.md (hunk 14 above)
Comment (revit13) [reply README.md:219]: Added Thanks

File: README.md (hunk 7 above)
Comment (roytman) [reply README.md:90]: > command to check the " `FybrikApplication` successfully deployed"

can you add the command to check that the `FybrikApplication` successfully deployed" ?

Comment (Mohammad-nassar10) [reply README.md:90]: In this step a `FybrikModule` is deployed not a `FybrikApplication`.

**Review by roytman — APPROVED** [review]
